finality-aleph/src/new_network/service.rs

```diff
@@ -165,22 +171,425 @@ impl<N: Network, D: Data> Service<N, D> {
                         return;
                     }
                 },
-                Some(maybe_ready) = async {
-                    match &maybe_sender {
-                        Some(sender) => Some(sender.ready().await),
-                        None => None,
-                    }
-                } => {
-                    match self.to_send.pop_front() {
-                        Some((data, peer, _)) => {
-                            if maybe_ready.ok().map(|ready| ready.send(data.encode()).ok()).flatten().is_none() {
-                                debug!(target: "aleph-network", "Failed sending data to peer {:?}", peer);
-                            }
-                        },
-                        None => warn!(target: "aleph-network", "Attempted to send data despite empty queue."),
-                    }
-                },
+                Some(()) =  Self::send(&self.network, &mut self.to_send) => { },
```

Comment:
in case we want to save one character
```suggestion
                Some(_) =  Self::send(&self.network, &mut self.to_send) => { },
```